From 9b1137430f29596fd609ddd75663e8a228699a81 Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@postmarketos.org>
Date: Sat, 6 Dec 2025 22:56:16 +0100
Subject: [PATCH] pgfutils: Improve byteorder detection

While s390x and SPARC are big endian, not all PowerPC targets are. Just
rely on the preprocessor macros to detect big endian targets instead.

Signed-off-by: Jens Reidel <adrian@postmarketos.org>
---
 core/libs/pgfutils/libpgf/PGFplatform.h | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/core/libs/pgfutils/libpgf/PGFplatform.h b/core/libs/pgfutils/libpgf/PGFplatform.h
index ac0708e9e7..1b7ad963a4 100644
--- a/core/libs/pgfutils/libpgf/PGFplatform.h
+++ b/core/libs/pgfutils/libpgf/PGFplatform.h
@@ -37,19 +37,7 @@
 // Endianess detection taken from lcms2 header.
 // This list can be endless, so only some checks are performed over here.
 //-------------------------------------------------------------------------------
-#if defined(_HOST_BIG_ENDIAN) || defined(__BIG_ENDIAN__) || defined(WORDS_BIGENDIAN)
-#define PGF_USE_BIG_ENDIAN 1
-#endif
-
-#if defined(__sgi__) || defined(__sgi) || defined(__powerpc__) || defined(__sparc) || defined(__sparc__)
-#define PGF_USE_BIG_ENDIAN 1
-#endif
-
-#if defined(__ppc__) || defined(__s390__) || defined(__s390x__)
-#define PGF_USE_BIG_ENDIAN 1
-#endif
-
-#ifdef TARGET_CPU_PPC
+#if defined(_HOST_BIG_ENDIAN) || defined(__BIG_ENDIAN__) || defined(WORDS_BIGENDIAN) || (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
 #define PGF_USE_BIG_ENDIAN 1
 #endif
 
-- 
2.52.0

